api/src/main/java/com/github/retrooper/packetevents/protocol/packettype/PacketType.java

```diff
@@ -806,187 +859,187 @@ public PacketSide getSide() {
-            BUNDLE,
-            CHUNK_BIOMES,
+            DAMAGE_EVENT(WrapperPlayServerDamageEvent.class),
+            HURT_ANIMATION(WrapperPlayServerHurtAnimation.class),
+            BUNDLE(WrapperPlayServerBundle.class),
+            CHUNK_BIOMES(null),
 
             // Added in 1.20.2
-            CHUNK_BATCH_END,
-            CHUNK_BATCH_BEGIN,
-            DEBUG_PONG,
-            CONFIGURATION_START,
+            CHUNK_BATCH_END(WrapperPlayServerChunkBatchEnd.class),
+            CHUNK_BATCH_BEGIN(WrapperPlayServerChunkBatchBegin.class),
+            DEBUG_PONG(WrapperPlayServerDebugPong.class),
+            CONFIGURATION_START(WrapperPlayServerConfigurationStart.class),
 
             // Added in 1.20.3
-            RESET_SCORE,
-            RESOURCE_PACK_REMOVE,
-            TICKING_STATE,
-            TICKING_STEP,
+            RESET_SCORE(WrapperPlayServerResetScore.class),
+            RESOURCE_PACK_REMOVE(WrapperPlayServerResourcePackRemove.class),
+            TICKING_STATE(WrapperPlayServerTickingState.class),
+            TICKING_STEP(WrapperPlayServerTickingStep.class),
 
             // Added in 1.20.5
-            COOKIE_REQUEST,
-            DEBUG_SAMPLE,
-            STORE_COOKIE,
-            TRANSFER,
-            PROJECTILE_POWER,
+            COOKIE_REQUEST(WrapperPlayServerCookieRequest.class),
+            DEBUG_SAMPLE(WrapperPlayServerDebugSample.class),
+            STORE_COOKIE(WrapperPlayServerStoreCookie.class),
+            TRANSFER(WrapperPlayServerTransfer.class),
+            PROJECTILE_POWER(WrapperPlayServerProjectilePower.class),
 
             // added in 1.21
-            CUSTOM_REPORT_DETAILS,
-            SERVER_LINKS,
+            CUSTOM_REPORT_DETAILS(WrapperPlayServerCustomReportDetails.class),
+            SERVER_LINKS(WrapperPlayServerServerLinks.class),
 
             // added in 1.21.2
-            MOVE_MINECART,
-            SET_CURSOR_ITEM,
-            SET_PLAYER_INVENTORY,
-            ENTITY_POSITION_SYNC,
-            PLAYER_ROTATION,
-            RECIPE_BOOK_ADD,
-            RECIPE_BOOK_REMOVE,
-            RECIPE_BOOK_SETTINGS,
+            MOVE_MINECART(WrapperPlayServerMoveMinecart.class),
+            SET_CURSOR_ITEM(WrapperPlayServerSetCursorItem.class),
+            SET_PLAYER_INVENTORY(WrapperPlayServerSetPlayerInventory.class),
+            ENTITY_POSITION_SYNC(WrapperPlayServerEntityPositionSync.class),
+            PLAYER_ROTATION(WrapperPlayServerPlayerRotation.class),
+            RECIPE_BOOK_ADD(WrapperPlayServerRecipeBookAdd.class),
+            RECIPE_BOOK_REMOVE(WrapperPlayServerRecipeBookRemove.class),
+            RECIPE_BOOK_SETTINGS(WrapperPlayServerRecipeBookSettings.class),
             ;
 
             private static int INDEX = 0;
             private static final Map<Byte, Map<Integer, PacketTypeCommon>> PACKET_TYPE_ID_MAP = new HashMap<>();
             private final int[] ids;
 
-            Server() {
+            Server(@Nullable Class<?> clazz) {
```

Comment:
```suggestion
            Server(@Nullable Class<? extends PacketWrapper<?>> clazz) {
```

Also, this doesn't seem to be saved + exposed as a field